plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/OTelResourceProvider.java

```diff
@@ -54,7 +53,7 @@ public static OpenTelemetrySdk get(TelemetrySettings telemetrySettings, Settings
                 settings,
                 OTelSpanExporterFactory.create(settings),
                 ContextPropagators.create(W3CTraceContextPropagator.getInstance()),
-                Sampler.parentBased(new RequestSampler(new ProbabilisticSampler(telemetrySettings)))
+                Sampler.parentBased(new RequestSampler(telemetrySettings))
```

Comment:
Since we have a number of samplers now, we need a way for the user to configure them: please add new setting to OTelTelemetrySettings (fe `telemetry.otel.tracer.span.sampler.classes`)
```
telemetry.otel.tracer.span.sampler.classes=...
```

For example, it could be used this way:

- use `ProbabilisticSampler` only (current behaviour):
```
telemetry.otel.tracer.span.sampler.classes=org.opensearch.telemetry.tracing.sampler.ProbabilisticSampler
```

- use `TransportActionSampler` only:
```
telemetry.otel.tracer.span.sampler.classes=org.opensearch.telemetry.tracing.sampler.TransportActionSampler
```

-  use `TransportActionSampler`  fronted by `ProbabilisticSampler` 
```
telemetry.otel.tracer.span.sampler.classes=org.opensearch.telemetry.tracing.sampler.ProbabilisticSampler, org.opensearch.telemetry.tracing.sampler.TransportActionSampler
```

This is what we've  discussed as delegation model.